Mathlib/Data/Bitvec/Lemmas.lean

```diff
@@ -27,6 +27,12 @@ theorem toNat_injective {n : Nat} : Function.Injective (@Std.BitVec.toNat n)
 theorem toNat_inj {x y : BitVec w} : x.toNat = y.toNat ↔ x = y :=
   toNat_injective.eq_iff
 
+theorem toFin_injective {n : Nat} : Function.Injective (@Std.BitVec.toFin n)
+  | ⟨_, _⟩, ⟨_, _⟩, rfl => rfl
+
+theorem toFin_inj {x y : BitVec w} : x.toFin = y.toFin ↔ x = y :=
+  toFin_injective.eq_iff
+
```

Comment:
nit: I'd put these before the `toNat` ones, as they're in some sense more fundamental. You could also then replace the proof of `Function.Injective (@Std.BitVec.toNat n)` with `toFin_injective.comp Fin.val_injective` or similar if you wanted, not that it really matters.